### PR TITLE
Allow extra build args

### PR DESCRIPTION
**Pull request by ofek**
e.g. `build_gcsfuse ${GOPATH}/src/github.com/googlecloudplatform/gcsfuse /tmp ${gcsfuse_version} "-ldflags=-s -ldflags=-w"`

@bjornleffler Hello there! If this looks alright to you, can we merge sometime tomorrow? I have need of this for a project 😄

### CODE REVIEW

**Comment by ofek** [comment]
Or possibly @bramp / @jacobsa if you have time please!

**Comment by bjornleffler** [comment]
I'll have a look later today.

**Comment by ofek** [comment]
@bjornleffler Awesome, thanks!

**Comment by ofek** [comment]
@bjornleffler This is the project btw https://github.com/ofek/docker-volume-gcs

**Comment by ofek** [comment]
@bjornleffler Are you alright with the general idea, just not implementation?

**Comment by bjornleffler** [comment]
Overall, this seems fine. I'd prefer for the extra build arguments to be on their own:
build_gcsfuse ... "-ldflags=-s" "-ldflags=-w"
So the usage becomes:
build_gcsfuse src_dir dst_dir version [ADDITIONAL_BUILD_FLAGS]

My only (small) concern is what happens if we'd like to add more arguments in the future, but let's deal with that when we get there.

Sorry for the late reply, I've had a busy week.

File: tools/build_gcsfuse/main.go [hunk 1]
```diff
@@ -45,17 +45,15 @@ import (
...
 // Build release binaries according to the supplied settings, setting up the
 // the file system structure we desire (see package-level comments).
 //
 // version is the gcsfuse version being built (e.g. "0.11.1"), or a short git
 // commit name if this is not for an official release.
-func buildBinaries(
-	dstDir string,
-	srcDir string,
-	version string) (err error) {
+func buildBinaries(dstDir, srcDir, version, buildArgs string) (err error) {
```
Comment (bjornleffler) [inline tools/build_gcsfuse/main.go]: Make buildArgs a []string will simplify the code below.

File: tools/build_gcsfuse/main.go [hunk 2]
```diff
@@ -214,17 +216,22 @@ func copyFile(dst string, src string, perm os.FileMode) (err error) {
 func run() (err error) {
 	// Extract arguments.
 	args := flag.Args()
-	if len(args) != 3 {
+	numArgs := len(args)
+	if numArgs != 3 && numArgs != 4 {
```
Comment (bjornleffler) [inline tools/build_gcsfuse/main.go]: if numArgs < 3

File: tools/build_gcsfuse/main.go [hunk 3]
```diff
@@ -214,17 +216,22 @@ func copyFile(dst string, src string, perm os.FileMode) (err error) {
...
 		err = fmt.Errorf("Usage: %s src_dir dst_dir version", os.Args[0])
 		return
 	}
 
 	srcDir := args[0]
 	dstDir := args[1]
 	version := args[2]
+	buildArgs := ""
```
Comment (bjornleffler) [inline tools/build_gcsfuse/main.go]: Make buildArgs an []string with args[3+].

**Comment by ofek** [comment]
@bjornleffler Done, thanks!

**Comment by ofek** [comment]
@bjornleffler How does this look now?

**Comment by ofek** [comment]
Thank you!!!

**Pull request merged**